src/push-manifest.ts

```diff
@@ -41,72 +81,106 @@ const extensionToTypeMapping: {[extname: string]: LinkType} = {
   '.woff': 'font',
   '.woff2': 'font'
 };
-export interface PushManifestEntry {
-  [dependencyAbsoluteUrl: string]: {
-    type?: LinkType,
-    // NOTE(fks) 04-05-2017: Only weight=1 is supported by browsers at the
-    // moment. When support is added, we can add automatic weighting and support
-    // multiple numbers.
+
+/**
+ * Get the default resource type for a file based on its extension.
+ */
+function getResourceTypeFromUrl(url: string): ResourceType|undefined {
+  return extensionToTypeMapping[path.extname(url)] || undefined;
+}
+
+/**
+ * Get the resource type for an import, handling special import types and
+ * falling back to getResourceTypeFromUrl() if none exist.
+ */
+function getResourceTypeFromImport(importFeature: Import): ResourceType|
+    undefined {
+  const importTagName = importFeature.astNode.tagName;
```

Comment:
I think you shouldn't have to check the ast node at all. `html-style` and `css-import` tells you that the file is css, `html-import` tells you that it's an html document, `html-script` tells you that it's a javascript file. `js-import` is a bit more interesting, amusingly, and there we should probably check the filename because an ES6 import's semantics hinge on the mime type it's served with. The others though, I think believe that they're blindly trusted to be the expected content type, so we can use the same semantics.